addons/planner/static/src/js/planner.js

```diff
@@ -284,6 +287,19 @@
             }
             self.update_planner(page_id);
         },
+        on_modal_hide: function(ev) {
+            this.save_input();
+        },
+        on_click_link: function(ev) {
+            $('#PlannerModal').modal('hide');
+        },
+        /*this method will save the data of all input elements when user close the modal,
+         or when user click on backend link */
+        save_input: function() {
+            var vals = this._get_values(false);
+            this.planner.data = _.extend(this.planner.data, vals);
+            this._save_planner_data();
+        }
```

Comment:
Instead of adding a class, using the openerp.web.bus to know when an action is triggered, the code will be simpler : http://pastie.org/10022544
Please have a look ;)